src/utils/query.ts

```diff
@@ -76,22 +79,54 @@ export const getColumnType = (type: string) => {
     }
 };
 
-/** parse response result field from ArrayRow to KeyValueRow */
+/** parse response result from ArrayRow to KeyValueRow */
+const parseModernResult = (rows: ArrayRow[], columns: ColumnType[]) => {
+    return rows.map((row) => {
+        return row.reduce<KeyValueRow>((newRow, cellData, columnIndex) => {
+            const {name} = columns[columnIndex];
+            newRow[name] = cellData;
+            return newRow;
+        }, {});
+    });
+};
+
 const parseExecuteModernResponse = (data: ExecuteModernResponse): IQueryResult => {
     const {result, columns, ...restData} = data;
 
     return {
-        result:
-            result &&
-            columns &&
-            result.map((row) => {
-                return row.reduce((newRow, cellData, columnIndex) => {
-                    const {name} = columns[columnIndex];
-                    newRow[name] = cellData;
-                    return newRow;
-                }, {} as KeyValueRow);
-            }),
+        result: result && columns && parseModernResult(result, columns),
         columns,
+        schema: 'modern',
```

Comment:
It seems that you don't need this field. In the end you are interested only in quantity of results. So in case of modern schema you may return array with only one result.

Reply:
Function `parseQueryAPIExecuteResponse` is also used in 7 other places. It such components as `Preview`, `TopShards`, `TopQueries`, `TenantOverview`, column tables overview. To make result always an array, it's also needed to support this format for all components, and for all schemas - `classic` and `ydb` - it's currently not parsed at all (`classic` is default when schema is not defined). So I found it easier to add additional filed to begin with.

As for the field, I'll delete it, you're right